### PR TITLE
[4.10] Build separate oc binaries for rhel 7 and rhel 8

### PR DESCRIPTION
**Pull request by mrunalp**
Build separate oc binaries for RHEL 7 and RHEL 8. Keep the native binary, in this case the RHEL 8 oc binary at /usr/bin/oc and 
/usr/share/openshift/linux_arm64/oc. Name the non-native binary with a suffix to distinguish.

@jupierce 
cc: @cgwalters @soltysh

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/label backport-risk-assessed

**Comment by jupierce** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oc/1488/pull-ci-openshift-oc-release-4.10-e2e-aws/1672624793045700608

```
ns/openshift-console pod/downloads-558f6b796f-67gs4 node/ip-10-0-155-75.us-west-2.compute.internal container/download-server restarted 6 times:
reason/ContainerExit code/1 cause/Error serving from /tmp/tmp2hixw7ea
Traceback (most recent call last):
  File "/tmp/serve.py", line 66, in <module>
    tar.add(path, basename)
  File "/usr/lib64/python3.6/tarfile.py", line 1938, in add
    tarinfo = self.gettarinfo(name, arcname)
  File "/usr/lib64/python3.6/tarfile.py", line 1807, in gettarinfo
    statres = os.lstat(name)
FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/openshift/linux_amd64/oc'
```

File: images/cli-artifacts/Dockerfile.rhel [hunk 1]
```diff
@@ -5,16 +5,21 @@ WORKDIR /go/src/github.com/openshift/oc
...
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/windows_amd64/oc.exe /usr/share/openshift/windows/oc.exe
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/share/openshift/linux_amd64/oc
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/bin/
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_arm64/oc /usr/share/openshift/linux_arm64/oc
+COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/share/openshift/linux_amd64/oc.rhel7
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_ppc64le/oc /usr/share/openshift/linux_ppc64le/oc
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_s390x/oc /usr/share/openshift/linux_s390x/oc
 COPY --from=builder /go/src/github.com/openshift/oc/LICENSE /usr/share/openshift/LICENSE
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/oc/oc /usr/bin/oc
+COPY --from=builder-rhel-8 /go/src/github.com/openshift/oc/oc /usr/share/openshift/linux_arm64/oc
```
Comment (jupierce) [inline images/cli-artifacts/Dockerfile.rhel:22]:
```suggestion
COPY --from=builder-rhel-8 /go/src/github.com/openshift/oc/oc /usr/share/openshift/linux_amd64/oc
```

**Comment by jupierce** [comment]
```
Exit code/1 cause/Error serving from /tmp/tmp_n4lgu8m\nTraceback (most recent call last):\n  File "/tmp/serve.py", line 66, in <module>\n    tar.add(path, basename)\n  File "/usr/lib64/python3.6/tarfile.py", line 1938, in add\n    tarinfo = self.gettarinfo(name, arcname)\n  File "/usr/lib64/python3.6/tarfile.py", line 1807, in gettarinfo\n    statres = os.lstat(name)\nFileNotFoundError: [Errno 2] No such file or directory: '/usr/share/openshift/linux_arm64/oc
```

File: images/cli-artifacts/Dockerfile.rhel [hunk 2]
```diff
@@ -5,16 +5,21 @@ WORKDIR /go/src/github.com/openshift/oc
...
+RUN make oc
+
 FROM registry.ci.openshift.org/ocp/4.10:cli
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/darwin_amd64/oc /usr/share/openshift/mac/oc
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/darwin_arm64/oc /usr/share/openshift/mac_arm64/oc
 COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/windows_amd64/oc.exe /usr/share/openshift/windows/oc.exe
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/share/openshift/linux_amd64/oc
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/bin/
-COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_arm64/oc /usr/share/openshift/linux_arm64/oc
+COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/share/openshift/linux_amd64/oc.rhel7
```
Comment (jupierce) [inline images/cli-artifacts/Dockerfile.rhel:17]:
```suggestion
COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_amd64/oc /usr/share/openshift/linux_amd64/oc.rhel7
COPY --from=builder /go/src/github.com/openshift/oc/_output/bin/linux_arm64/oc /usr/share/openshift/linux_arm64/oc
```

**Pull request merged**